cmake/thirdparty/SetupCUDA.cmake

```diff
@@ -45,8 +45,19 @@
 ################################
 set (CMAKE_MODULE_PATH "${BLT_ROOT_DIR}/cmake/thirdparty;${CMAKE_MODULE_PATH}")
 
+
+if ( NOT CMAKE_CUDA_HOST_COMPILER )
+    if ( CMAKE_CXX_COMPILER )
+        set ( CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER} CACHE STRING "" FORCE)
+    else ()
+        set ( CMAKE_CUDA_HOST_COMPILER ${CMAKE_C_COMPILER} CACHE BOOL "" FORCE)
+    endif ()
```

Comment:
It may be helpful to add `message()` here to spit out the value of `CMAKE_CUDA_HOST_COMPILER`